### ISSUE TITLE
frequent unpredictable panics

### ISSUE THREAD

**Issue by necro351** [andlabs/ui#75]
Hi,

It seems the unsafe pointer for the window keeps getting zapped within the API when I do non-trivial things in the event handlers for buttons. I have a simple application that uses the net/http library to GET/POST things depending on what button is pressed. After using these buttons the UI will crash when being resized, or if other buttons are pressed with messages like this:

panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x403b612]

goroutine 1 [running, locked to thread]:
github.com/andlabs/ui.(*event).fire(0x0, 0x4385d00)
	/Users/rick/git/go/src/github.com/andlabs/ui/uitask.go:106 +0x82
github.com/andlabs/ui.buttonClicked(0x1591c408)
	/Users/rick/git/go/src/github.com/andlabs/ui/button_darwin.go:46 +0x25
github.com/andlabs/ui._Cfunc_uimsgloop()
	/Users/rick/git/go/src/github.com/andlabs/ui/:442 +0x39
github.com/andlabs/ui.uimsgloop()
	/Users/rick/git/go/src/github.com/andlabs/ui/uitask_darwin.go:27 +0x18
github.com/andlabs/ui.Go(0x0, 0x0)
	/Users/rick/git/go/src/github.com/andlabs/ui/uitask.go:22 +0x63
main.main()
	/Users/rick/git/go/src/storytime/ui/ui.go:28 +0x56

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_386.s:2287 +0x1

goroutine 6 [chan receive]:
github.com/andlabs/ui.uiissueloop()
	/Users/rick/git/go/src/github.com/andlabs/ui/uitask.go:59 +0x44
created by github.com/andlabs/ui.Go
	/Users/rick/git/go/src/github.com/andlabs/ui/uitask.go:21 +0x5c
Harvey%

**Comment by necro351** [comment]
Thought I would mention that my panics go away when I run with GOGC=off. Then I remembered you said windows must be declared as globals to avoid GC, and I realized I mistakenly (automatically rather) declared it as a local in a function. Once I moved it out to a global and made sure to assign to that, the panics went away (even without GOGC=off).

That said, it really sucks that ui.Windows must be declared globally, it seems very very brittle.
